### PR TITLE
Fix races in executor, zoo and healthchecker packages

### PR DESCRIPTION
**Pull request by s-urbaniak**

### CODE REVIEW

**Comment by tsenart** [comment]
While killing all these races down, @s-urbaniak also fixed some in our test library: https://github.com/stretchr/testify/pull/177

**Comment by tsenart** [comment]
PTAL

**Comment by tsenart** [comment]
LGTM

File: executor/executor.go [hunk 1]
```diff
@@ -345,6 +350,7 @@ func (driver *MesosExecutorDriver) statusUpdateAcknowledgement(from *upid.UPID,
 
 	// Remove the corresponding update.
 	delete(driver.updates, uuid.String())
+	driver.lock.Unlock()
```
Comment (jdef) [inline executor/executor.go]: should we unlock, perhaps, after the next line that deletes from driver.tasks?

Comment (tsenart) [inline executor/executor.go]: SGTM

**Comment by jdef** [comment]
minor nit, otherwise LGTM

**Pull request merged**